src/utils/material-utils.js

```diff
@@ -240,5 +255,68 @@ export function disposeTexture(texture) {
     texture.hls = null;
   }
 
+  if (originalTextureUuidMap.has(texture)) {
+    originalTextureUuidMap.delete(texture);
+    resizedTextureCache.delete(originalTextureUuidMap.get(texture));
+  }
+
   texture.dispose();
 }
+
+const resizedTextureCache = new Map();
+const originalTextureUuidMap = new WeakMap();
+function resizeTexture(texture, width, height) {
+  if (!texture) {
+    return texture;
+  }
+
+  if (resizedTextureCache.has(texture.uuid)) {
+    return resizedTextureCache.get(texture.uuid);
+  }
+
+  // Ignore compressed textures and video textures for now.
+  if (texture.isCompressedTexture || texture.isVideoTexture ) {
+    return texture;
+  }
+
+  const image = texture.image;
+
+  // Just in case.
+  if (!image || image.width === undefined || image.height === undefined) {
+    return texture;
+  }
+
+  if (image.width <= width && image.height <= height) {
+    return texture;
+  }
+
+  const newWidth = Math.min(width, image.width);
+  const newHeight = Math.min(height, image.height);
```

Comment:
Consider keep the aspectRatio here based on width / height?

Use case: we reused this function to resize some large media-image to avoid crash the safari.

Reply:
Sounds like a good idea. Probably I will do it if we introduce this feature, thanks for suggestion.

Reply:
Done